### ISSUE TITLE
Named coerce functions not available in Cerberus 0.9

### ISSUE THREAD

**Issue by sybrenstuvel** [pyeve/cerberus#369]
Used Cerberus version: 0.9.2 (pinned by Eve)

- I consulted these documentations:
  - [x] http://docs.python-cerberus.org/en/stable/
  - [x] http://docs.python-cerberus.org/en/latest/

- I consulted these sections of the docs (add more lines as necessary):
  - http://docs.python-cerberus.org/en/stable/customize.html#custom-coercers
  - http://docs.python-cerberus.org/en/stable/normalization-rules.html#value-coercion

----

#### Use-case abstract

I want to add custom coercion rules by implementing a `_normalize_coerce_{rulename}` function.

----

#### Support request / Bug report

The documentation, more specifically the section on [custom coercers](http://docs.python-cerberus.org/en/stable/customize.html#custom-coercers), mention that this feature was introduced in Cerberus 0.9. The other section in the documentation on this subject ([Value Coercion](http://docs.python-cerberus.org/en/stable/normalization-rules.html#value-coercion)) doesn't mention any version in which this was introduced.

Implementing the `_normalize_coerce_multiply` example causes a problem, where the value `2` cannot be coerced. Inspecting the [source code of Cerberus](https://github.com/pyeve/cerberus/blob/dabc063c32d14738d3abadc0b7b82d1a98f41c2c/cerberus/cerberus.py#L453) shows that it doesn't do any by-name lookup on the validator class at all.

It would seem that using the name of a custom normalization method is not supported in 0.9, and my guess is that it was introduced in 1.0. However, the introduction of this isn't even mentioned in [the changelog](http://docs.python-cerberus.org/en/stable/changelog.html#version-1-0).

While 0.9.2 is pinned by Eve, please provide documentation for that version. Right now I'm loosing time struggling with Cerberus and trying get features to run that aren't supported in the first place. Note that the Eve documentation even points to http://docs.python-cerberus.org/en/latest/usage.html#type-coercion (note the `latest` in that URL), adding to the confusion.

**Comment by funkyfuture** [comment]
> However, the introduction of this isn't even mentioned in the changelog.

it's mentioned in `changelog_items[11]`

looks to me like an `eve` issue anyway.

**Comment by nicolaiarocci** [comment]
Eve 0.8 works with Cerberus 1.x... until it is released you might want to install from the development (master, actually) branch.

**Comment by sybrenstuvel** [comment]
@funkyfuture I don't know what `changelog_items[11]` refers to. Do you mean this line?

> New: ‘coerce’, ‘rename_handler’ and ‘validator’ can use class-methods (Frank Sachsenheim).

Class methods are methods decorated with `@classmethod`, so this doesn't refer to the issue I described. If "class-methods" are to be understood as "methods defined in a class", then the changelog uses different names for things than Python itself, which I assume you'll understand would cause all kind of confusion.

@nicolaiarocci It *is* an issue with Cerberus documentation, as a feature released in a later version (using a string to name a coercion function) is described as "New in version 0.9".

**Comment by sybrenstuvel** [comment]
I have this fixed in https://github.com/armadillica/cerberus/tree/sybren-0.9, I would send a pull requests if you'd have a branch for the 0.9 releases.

**Comment by funkyfuture** [comment]
you're right, `… can use validator methods` would be more correct. you may provide a patch for this.

i don't get the point about the documentation. 0.9 shipped with the first implementation of `coerce` that could only consume callables as arguments. there is no claim that it works as in 1.0.

**Comment by sybrenstuvel** [comment]
Take another look at the [Value Coercion](http://docs.python-cerberus.org/en/stable/normalization-rules.html#value-coercion) section in the documentation. It states:

> Coercion allows you to apply a callable (given as object or the name of a custom normalization method)

and

> New in version 0.9

It doesn't mention at all that those are actually two features:
- "given as object", which can be used in 0.9, and
- "the name of a custom normalization method", which actually requires 1.0+ (and thus is unusable with any current Eve release)

Simply stating "New in version 0.9" is misleading.

Of course I could have gotten the 0.9.2 sources from Github, installed Sphinx, built the documentation myself for that particular version, and read there again to double-check. However, if you claim that this feature is available in 0.9 onward...

**Comment by funkyfuture** [comment]
i see. for the 1.0 release i removed all `.. versionchanged` annotations from the docs as these haven't been complete and since a lot changed until 1.0 they'd been also rather confusing boilerplate informaion for the 1.0 users.

should all `.. versionadded` annotations be removed as well?

**Comment by sybrenstuvel** [comment]
Yes, of course! How does "introduced in 0.9" help when "changed in 1.0" is removed?

I would suggest removing all of those annotations, and making the documentation for the in-use versions (like the version used by the last Eve release) available with explicit version numbers.
